packages/react/src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+import ExpandCollapsePanel, { PanelTrigger } from '../ExpandCollapsePanel';
+import randomId from '../../utils/rndid';
+import { IconType } from '../Icon';
+
+interface AccordionProps extends React.HTMLAttributes<HTMLDivElement> {
+  className?: string;
+  children: React.ReactNode | React.ReactNode[];
+  props?: React.HTMLAttributes<HTMLDivElement>;
+}
+
+const Accordion = ({ className, children, ...props }: AccordionProps) => {
+  return (
+    <div
+      className={classNames(className ? className : '', 'Accordion__container')}
+      {...props}
+    >
+      {children}
+    </div>
+  );
+};
+
+interface AccordionContainerProps extends React.HTMLAttributes<HTMLDivElement> {
+  className?: string;
+  children: React.ReactElement[];
+  open?: boolean;
+  setIsOpen?: (open: boolean) => void;
+  trigger?: React.ReactNode | React.ReactNode[];
+  iconExpanded?: IconType;
+  iconCollapsed?: IconType;
+  shouldHideIcon?: boolean;
+  isControlled?: boolean;
+  animationTiming?: number | boolean;
+  props?: React.HTMLAttributes<HTMLDivElement>;
+}
+
+const AccordionContainer = ({
+  children,
+  open = false,
+  setIsOpen,
+  iconExpanded = 'triangle-down',
+  iconCollapsed = 'triangle-right',
+  shouldHideIcon = false,
+  isControlled = false,
+  animationTiming,
+  ...props
+}: AccordionContainerProps) => {
+  const [elementId, setElementId] = useState<string | null>(null);
+  const [notControlledOpen, setNotControlledOpen] = useState(false);
+
+  useEffect(() => {
+    setElementId(randomId());
+  }, []);
+
+  if (!isControlled) {
+    return (
+      <ExpandCollapsePanel
+        id={`${elementId}-panel`}
+        open={notControlledOpen}
+        onToggle={() => setNotControlledOpen(!notControlledOpen)}
+        aria-labelledby={`${elementId}-trigger`}
+        animationTiming={animationTiming}
+        {...props}
+      >
+        <PanelTrigger
+          id={`${elementId}-trigger`}
+          iconCollapsed={iconCollapsed}
+          iconExpanded={iconExpanded}
+          className={classNames(
+            children[0].props.className
+              ? children[0].props.className
+              : 'Accordion__trigger'
+          )}
```

Comment:
I'm a bit confused about what's happening here. What are you trying to accomplish?

Reply:
So this was to address the specific ordering of children that the underlying expand/collapse component requires. I tried several ways, including passing in a child component itself, but when the expand/collapse would not clone the elements in the correct order unless being explicit in this manner. 

It's also what ended up necessitating the container altogether, but perhaps this is something we could look at together? Could certainly use your input/advice on how to handle it!